python/src/space/core/storage.py

```diff
@@ -210,58 +233,102 @@ def _lookup_reference(self, tag_or_branch: str) -> meta.SnapshotReference:
 
   def add_tag(self, tag: str, snapshot_id: Optional[int] = None) -> None:
     """Add tag to a snapshot"""
+    self._add_reference(tag, meta.SnapshotReference.TAG, snapshot_id)
+
+  def add_branch(self, branch: str) -> None:
+    """Add branch to a snapshot"""
+    self._add_reference(branch, meta.SnapshotReference.BRANCH, None)
+
+  def set_current_branch(self, branch: str):
+    """Set current branch for the snapshot."""
+    if branch != "main":
+      snapshot_ref = self.lookup_reference(branch)
+      if snapshot_ref.type != meta.SnapshotReference.BRANCH:
+        raise errors.UserInputError("{branch} is not a branch.")
+    self._current_branch = branch
+
+  def _add_reference(
+      self,
+      reference_name: str,
```

Comment:
nit: maybe ref_name and ref_type to make code shorter? You have used ref_name in other method